### DISCUSSION TITLE
How to deploy MIKE1D SDK application so it could run without MIKE1D installation

### DISCUSSION THREAD

**Discussion by hawklorry** [DHI/Mike1D-SDK-examples#26] [Q&A] [answered]
We are developing a tool to export result summaries and want to deploy it as a standard .NET application. Users may not have MIKE1D installed. In the example, it recommends to check the MIKE Install as shown below. Is there a way to deploy the application without MIKE1D installation? We only read res1d files. Does it require MIKE licenses? 

`if (!DHI.Mike.Install.MikeImport.Setup(20, DHI.Mike.Install.MikeProducts.Mike1D))
                throw new Exception("Could not find a MIKE installation");`

**Comment by JesperGr** [comment, marked as answer]
If you only need to handle res1d files, then only the MIKE 1D `ResultDataAccess` and the `Generic` component are required. For that, everything needed is included in the `DHI.Mike1D.ResultDataAccess` NuGet package and its dependencies, which you can just grab and xcopy deploy together with your own tool. No license is required, no MIKE installation, and no use of `DHI.MIKE.Install`. 

The tool has to be 64 bit, and if building for .NET Core / .NET 5+ in Visual Studio, you need the Publish step to get all the native dependencies.

**Reply by hawklorry** [reply]
That's our intension. We just read the coordinates of the network and catchments. 

The program uses .NET framework 4.7.2. Following your instruction, I added DHI.Mike1D.ResultDataAccess (not sure the xcopy part though) and the follow error occur. The DHI DLLs included in the bin folder is also given below.  It seems the required native library is not added to the bin folder. Anything missing here? 

![image](https://user-images.githubusercontent.com/3594975/172860929-51bc641b-ce97-4c07-9716-09e3f9c43c7a.png)

![image](https://user-images.githubusercontent.com/3594975/172861753-e3318724-0e99-4850-b940-43d5b547d7a0.png)

**Reply by hawklorry** [reply]
I got through the previous error by copying the native dependencies. It now give a weird error shown below. Any thoughts?  

![image](https://user-images.githubusercontent.com/3594975/172865056-4a943d30-3e63-454f-a528-07bdd808b8c7.png)

**Comment by JesperGr** [comment]
If I make a .NET Framework console application, add `DHI.Mike1D.ResultDataAccess` as a NuGet reference, change the `Platform target` to `x64`, then in the bin debug/release folder I get the files as shown in the image below. 

Two details that makes this work: `Platform target` must be `x64`, and NuGet packages must be using `PackageReference`. For details, see [MIKE for Developers - NuGet troubleshooting](https://docs.mikepoweredbydhi.com/nuget/#troubleshooting)

![image](https://user-images.githubusercontent.com/25352297/172865992-2647fd17-b19b-4e98-9a88-e23c622cf447.png)

**Comment by JesperGr** [comment]
Platform target is in the project properties, and `x64` must be set for both debug and release:

![image](https://user-images.githubusercontent.com/25352297/172867460-8a47ca7b-f9c1-4052-a122-67de3dd2a023.png)

**Reply by hawklorry** [reply]
Thanks @JesperGr I have followed the steps as show below. DHI.Mike1D.ResultDataAccess is used in a class library which is referenced in a WPF application and test library. The native dependencies are generated in the class library bin folder but not copied to the WPF application and test library automatically. I'm thinking to use xcopy in the post-build event command line to copy these files. 

![image](https://user-images.githubusercontent.com/3594975/172875901-0f160ed3-addd-4769-9c7f-84fb9da96c63.png)

![image](https://user-images.githubusercontent.com/3594975/172875947-a8550967-8180-4107-9854-e1a7aa5975f9.png)

The second error was resolved. I only copied the dll files and forget some other files (ubg, pfs files). Now the system is working.
